aspnetcore/fundamentals/middleware/extensibility.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Middleware activation with nonconforming containers in ASP.NET Core
+author: guardrex
+description: Learn how to activate middleware with nonconforming containers in ASP.NET Core.
+ms.author: riande
+manager: wpickett
+ms.custom: mvc
+ms.date: 01/17/2018
+ms.topic: article
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: fundamentals/middleware/extensibility
+---
+# Middleware activation with nonconforming containers in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+[IMiddlewareFactory](/dotnet/api/microsoft.aspnetcore.http.imiddlewarefactory)/[IMiddleware](/dotnet/api/microsoft.aspnetcore.http.imiddleware) is an extensibility point for [middleware](xref:fundamentals/middleware/index) activation with nonconforming containers.
+
+`UseMiddleware` extension methods check if a middleware's registered type implements `IMiddleware`. If it does, the `IMiddlewareFactory` instance registered in the nonconforming container is used to resolve the `IMiddleware` implementation instead of using the convention-based middleware activation logic. The factory is registered as a scoped service with the app's noncomforming service container.
+
+Benefits:
+
+* Activation per request (injection of scoped services)
+* Automatic middleware activation with nonconforming containers
```

Comment:
I'm not sure we state what a non-conforming container is anywhere else so I don't know if this makes sense to mention.

Reply:
> Automatic middleware activation with 3rd party containers

You said 'that we don't have an adapter for' earlier when I suggested the list of containers that could be shown in the sample. We can't say "adapter" here either without explaining what that means.

Strike the remark, or should we flesh it out more specifically?